### PR TITLE
Avoid using too many Base Classes

### PR DESCRIPTION
**Pull request by sidsharma2002**
Too many Base classes can result in a deep issue which is necessary to avoid from the beginning. Hence I have added this point.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -255,6 +255,7 @@ Split your app module into different small modules, and give those modules as de
 
 <br/>Read more [here](https://medium.com/swlh/modularization-by-feature-and-layer-with-android-architecture-components-80bf317d737).
 
+28) Avoid using too many Base classes. Using base classes everywhere makes a tight web in your codebase, which later makes it hard to do changes in the code. If still necessary, create and use a standalone function (kotline file).
```
Comment (niharika2810) [inline README.md]: Hey Thank you for contributing. Could you please add a few example or details for anyone to undestand it better.

Comment (sidsharma2002) [reply README.md]: Hi, I have added a detailed explaination along with some references.

File: README.md [hunk 2]
```diff
@@ -255,6 +255,11 @@ Split your app module into different small modules, and give those modules as de
...
```
Comment (niharika2810) [inline README.md]: Hey this example looks perfect. A few spelling mistakes. Could you pleas correct it? I wll merge then. Thank you once again :)

Comment (sidsharma2002) [reply README.md]: Did some changes, please have a look and let me know.

**Review by niharika2810 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (niharika2810) [reply README.md]: Thank you for the contribution. Have merged it :)

**Pull request merged**